docroot/modules/custom/va_gov_menu_access/src/Service/MenuReductionService.php

```diff
@@ -0,0 +1,595 @@
+        $this->menuRules['universal_parent_menu_items'][] = trim(end($universal_parent_menu_items_raw));
+      }
+    }
+    // Grab the locked_paths config.
+    $locked_paths = $this->settings->get('va_gov_menu_access.locked_paths');
+    $locked_paths_array = explode("\n", $locked_paths);
+    // Sort locked paths into universal and single allowance arrays.
+    foreach ($locked_paths_array as $locked_path) {
+      if (strpos($locked_path, '%') !== FALSE) {
+        $universal_locked_raw = explode('%', $locked_path);
+        $this->menuRules['universal_locked_paths'][] = trim(end($universal_locked_raw));
+      }
+      else {
+        $this->menuRules['single_locked_paths'][] = trim($parent);
+      }
+    }
+  }
+
+  /**
+   * Extract menu ids and other date from parent options.
+   *
+   * @param mixed $parent_options
+   *   Array of options from the menu link.
+   *
+   * @return array
+   *   An array of menu ids and related metadata.
+   */
+  protected function extractMenuIds($parent_options) : array {
+    $uuids_with_vals = [];
+    $key_count = 0;
+    foreach ($parent_options as $key => $option) {
+      if (!empty(explode('menu_link_content:', $key))) {
+        // This is how we get our menu item uuid.
+        $menu_id_array = explode('menu_link_content:', $key);
+        if (!empty($menu_id_array[1])) {
+          $menu_id = $menu_id_array[1];
+          $uuids_with_vals[$menu_id] = [
+            'key_count' => $key_count++,
+            'menu_key' => $key,
+            'option' => $option,
+          ];
+        }
+      }
+    }
+    return $uuids_with_vals;
+  }
+
+  /**
+   * Used to close out a session of the menu reduction service so no persist.
+   */
+  protected function nuke() {
+    unset($this->formState);
+    $this->formState = NULL;
+    unset($this->nodeBundle);
+    $this->nodeBundle = NULL;
+    unset($this->currentMenuParent);
+    $this->currentMenuParent = '';
+    unset($this->originalMenuParentOptions);
+    $this->originalMenuParentOptions = [];
+  }
+
+  /**
+   * Checks to see if the menu parent would be changed, backs out if has.
+   *
+   * @param array $form
+   *   The form array by reference.
+   */
+  protected function preventUnintendedChangeOfParent(array &$form) {
+    // Check to see if the current parent exists in the reduced form.
+    if (!in_array($this->currentMenuParent, $form['menu']['link']['menu_parent']['#options'])) {
+      // Parent does not exist in reduced form, Put the original parent options
+      // back to prevent data loss. The existing menu setting should not be
+      // allowed, but it exists, so allow it to persist. It may have been
+      // created and approved by a content_admin.
+      $form['menu']['link']['menu_parent']['#options'] = $this->originalMenuParentOptions;
+      // The menu form itself should be hidden for non-admins.
+      $form['menu']['#attributes'] = ['style' => 'display:none'];
```

Comment:
@swirtSJW I'm curious why you are adding an inline style, in lieu of something like 
```suggestion
      $form['menu']['#attributes']['class'][] = 'element-invisible';
```

Reply:
Yah I hemmed and hawed on that one for a bit.  I mainly chose the inline style because I did not want a css snafu surfacing it.  I could be talked either way on it.  It felt dirty, but also felt right ;)